subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPluginOptions.kt

```diff
@@ -41,10 +43,24 @@ class KotlinDslPluginOptions internal constructor(objects: ObjectFactory) {
     }
 
     /**
-     * Set to `false` to silence the warning about the `kotlin-dsl` plugin enabling Kotlin compiler experimental features.
+     * Flag has no effect since `kotlin-dsl` no longer relies on experimental features.
      */
-    val experimentalWarning = objects.property<Boolean>().apply {
-        set(true)
+    @Deprecated("Flag has no effect since `kotlin-dsl` no longer relies on experimental features.")
+    val experimentalWarning: Property<Boolean>
+        get() {
+            nagUserAboutExperimentalWarning()
+            return experimentalWarningProperty
+        }
+
+    private
+    val experimentalWarningProperty = objects.property<Boolean>()
+
+    private
+    fun nagUserAboutExperimentalWarning() {
+        deprecate("kotlinDslPluginOptions.experimentalWarning")
```

Comment:
:x: This declares a deprecated _feature_. It should use `deprecateProperty(name)` instead.

Reply:
I went ahead and refined the deprecation message